webview/src/plots/components/SizeSliders.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useCallback } from 'react'
+import { useDispatch } from 'react-redux'
+import { PlotHeight, PlotsSection } from 'dvc/src/plots/webview/contract'
+import styles from './styles.module.scss'
+import { changeSize as changeTemplatePlotsSize } from './templatePlots/templatePlotsSlice'
+import { changeSize as changeCustomPlotsSize } from './customPlots/customPlotsSlice'
+import { changeSize as changeImagesSize } from './comparisonTable/comparisonTableSlice'
+import { Slider } from '../../shared/components/slider/Slider'
+import { resizePlots } from '../util/messages'
+
+interface SizeSlidersProps {
+  maxNbPlotsPerRow: number
+  height: number
+  sectionKey: PlotsSection
+  nbItemsPerRowOrWidth: number
+  noHeight?: boolean
+}
+
+export const SizeSliders: React.FC<SizeSlidersProps> = ({
+  maxNbPlotsPerRow,
+  height,
+  sectionKey,
+  nbItemsPerRowOrWidth,
+  noHeight
+}) => {
+  const dispatch = useDispatch()
+
+  const handleResize = useCallback(
+    (nbItems: number, newHeight: PlotHeight) => {
+      const changeSize = {
+        [PlotsSection.TEMPLATE_PLOTS]: changeTemplatePlotsSize,
+        [PlotsSection.COMPARISON_TABLE]: changeImagesSize,
+        [PlotsSection.CUSTOM_PLOTS]: changeCustomPlotsSize
+      }
+      const positiveNbItems = Math.abs(nbItems)
+      dispatch(
+        changeSize[sectionKey]({
+          height: newHeight,
+          nbItemsPerRowOrWidth: positiveNbItems
+        })
+      )
+      resizePlots(newHeight, positiveNbItems, sectionKey)
+    },
+    [dispatch, sectionKey]
+  )
+
+  const plotHeights = Object.values(PlotHeight).filter(
+    value => typeof value !== 'string'
```

Comment:
[Q] Why is this needed?

Reply:
If I remember correctly, `PlotHeight` is an enum:

```
export enum PlotHeight {
  SMALLER,
  SMALL,
  REGULAR,
  SQUARE,
  VERTICAL_NORMAL,
  VERTICAL_LARGER
}
```
which means that `Object.values(PlotHeight)` will return `[0, 1, 2, 3, 4, 5, 'SMALLER', 'SMALL',...]`. This removes the string values as they duplicate the numerical values.